spec/features/results/response_form_tree_handling_spec.rb

```diff
@@ -94,8 +100,92 @@
+      fill_in("response_root_children_0_children_0_value", with: "1")
+      drop_in_dropzone(Rails.root.join("spec", "fixtures", "media", "audio", "powerup.mp3"))
+      expect(page).to have_content("The uploaded file was not an accepted format.")
+      drop_in_dropzone(Rails.root.join("spec", "fixtures", "media", "images", "the_swing.jpg"))
+      select("Animal")
+      select("Dog")
+      fill_in("response_root_children_3_children_0_children_0_children_0_value", with: "4561")
+      fill_in("response_root_children_3_children_0_children_1_children_0_children_0_value", with: "7891")
+      all("a.add-repeat").last.click
+      fill_in("response_root_children_3_children_1_children_0_children_0_value", with: "4562")
+      fill_in("response_root_children_3_children_1_children_1_children_0_children_0_value", with: "7892")
+      click_button("Save")
+
+      expect(page).to have_content("Response is invalid")
+      expect_value("#response_root_children_3_children_0_children_0_children_0_value", "4561")
+      expect_value("#response_root_children_3_children_0_children_1_children_0_children_0_value", "7891")
+      expect_value("#response_root_children_3_children_1_children_0_children_0_value", "4562")
+      expect_value("#response_root_children_3_children_1_children_1_children_0_children_0_value", "7892")
+
+      fill_in("response_root_children_0_children_0_value", with: "123")
+      click_button("Save")
+
+      expect(page).to_not have_content("Response is invalid")
+
+      response = Response.last
+      visit hierarchical_response_path(params.merge(id: response.shortcode))
+
+      expect(page).to have_content("123")
+      expect(page).to have_selector("[data-qtype-name=image] .media-thumbnail img")
+      expect(page).to have_content("Animal")
+      expect(page).to have_content("Dog")
+      expect(page).to have_content("4561")
+      expect(page).to have_content("7891")
+      expect(page).to have_content("4562")
+      expect(page).to have_content("7892")
+    end
+
+    context "with conditional logic" do
+      before do
+        ref_qing = form.root_group.c[0].c[0]
+
+        questioning = form.root_group.c[2]
+        questioning.display_if = "all_met"
+        questioning.display_conditions_attributes = [
+          {ref_qing_id: ref_qing.id, op: "eq", value: "123"}
+        ]
+        questioning.save!
+      end
+
+      scenario "submitting response" do
+        visit new_hierarchical_response_path(params)
+
+        select2(user.name, from: "response_user_id")
+
+        # makes select boxes visible
+        fill_in("response_root_children_0_children_0_value", with: "123")
+
+        select("Animal")
+        select("Dog")
+
+        # hides select boxes, they are now irrelevant
+        fill_in("response_root_children_0_children_0_value", with: "124")
+
+        click_button("Save")
+
+        response = Response.last
+        visit hierarchical_response_path(params.merge(id: response.shortcode))
```

Comment:
Let's please use edit and expect_value etc. to test that this works since we're not supporting `#show` yet.